libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginBaseFormFragment.java

```diff
@@ -273,45 +266,19 @@ protected void onLoginFinished(boolean success) {
         }
     }
 
-    protected void startPostLoginServices() {
-        // Get reader tags so they're available as soon as the Reader is accessed - done for
-        // both wp.com and self-hosted (self-hosted = "logged out" reader) - note that this
-        // uses the application context since the activity is finished immediately below
-        ReaderUpdateService.startService(getActivity().getApplicationContext(), EnumSet.of(ReaderUpdateService
-                .UpdateTask.TAGS));
-
-        // Start Notification service
-        NotificationsUpdateService.startService(getActivity().getApplicationContext());
-    }
-
-    protected void saveCredentialsInSmartLock(SmartLockHelper smartLockHelper, String username, String password) {
+    protected void saveCredentialsInSmartLock(LoginListener loginListener, String username, String password) {
         // mUsername and mPassword are null when the user log in with a magic link
-        if (smartLockHelper != null) {
-            smartLockHelper.saveCredentialsInSmartLock(username, password,
-                    mAccountStore.getAccount().getDisplayName(),
+        if (loginListener != null) {
+            loginListener.saveCredentials(username, password, mAccountStore.getAccount().getDisplayName(),
```

Comment:
Sure, let's use `saveCredentialsInSmartLock`.

I think my intention here had been to keep things implementation-independent, since the 'host app' could be storing those credentials in some way other than smart lock. But a better solution would have been a 'callback'-type method (`somethingHappened`) rather than `doThis`. I didn't see a good naming scheme that would make things clearer rather than more obscure, and in this case naming smart lock specifically is probably clearer. 👍

Reply:
Done in 60cd9e1.

Reply:
Ah, I see the rationale there, yes. Interestingly, the "callback"-type like method is pretty much what I did in the Service-based login procedure [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginWpcomService.java#L290), where an event is emitted to signal that the credentials were accepted, and the client can decide what to do in that case.